pkg/virt-controller/services/multus_annotations.go

```diff
@@ -0,0 +1,81 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021 Red Hat, Inc.
+ *
+ */
+
+package services
+
+import (
+	"encoding/json"
+	"fmt"
+
+	v1 "kubevirt.io/client-go/api/v1"
+)
+
+type multusNetworkAnnotation struct {
+	InterfaceName string `json:"interface"`
+	Mac           string `json:"mac,omitempty"`
+	NetworkName   string `json:"name"`
+	Namespace     string `json:"namespace"`
+}
+
+func getMultusCNIAnnotation(vmi *v1.VirtualMachineInstance) (string, error) {
+	var multusNetworksAnnotationStruct []multusNetworkAnnotation
+
+	multusNonDefaultNetworks := filterMultusNonDefaultNetworks(vmi)
+	for i, network := range multusNonDefaultNetworks {
+		serializedInterface := newMultusAnnotationStruct(vmi, network, fmt.Sprintf("net%d", i+1))
+		multusNetworksAnnotationStruct = append(multusNetworksAnnotationStruct, serializedInterface)
+	}
+
+	if len(multusNetworksAnnotationStruct) > 0 {
+		multusNetworksAnnotation, err := json.Marshal(multusNetworksAnnotationStruct)
+		if err != nil {
+			return "", fmt.Errorf("failed to create JSON list from CNI interface map %v", multusNetworksAnnotationStruct)
+		}
+		return string(multusNetworksAnnotation), nil
+	}
+	return "", nil
+}
+
+func newMultusAnnotationStruct(vmi *v1.VirtualMachineInstance, network v1.Network, podInterfaceName string) multusNetworkAnnotation {
+	multusIface := getIfaceByName(vmi, network.Name)
```

Comment:
I think you can move here `getIfaceByName` as well.

Reply:
Done.